compiler/noirc_evaluator/src/ssa/opt/mem2reg.rs

```diff
@@ -252,15 +252,15 @@ impl<'f> PerFunctionContext<'f> {
                     return true;
                 }
 
-                let all_aliases_set_for_removal = aliases.any(|alias| {
+                let has_alias_not_marked_for_removal = aliases.any(|alias| {
```

Comment:
Could we just add a comment above here? Something like "Check whether there are any aliases whose instructions are not all marked for removal. If there is any alias marked to survive, we should not remove its last store."